src/lib/tuning/index/index_operation.cpp

```diff
@@ -0,0 +1,85 @@
+#include "index_operation.hpp"
+
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "sql/sql_query_cache.hpp"
+#include "sql/sql_query_plan.hpp"
+#include "storage/index/adaptive_radix_tree/adaptive_radix_tree_index.hpp"
+#include "storage/index/group_key/composite_group_key_index.hpp"
+#include "storage/index/group_key/group_key_index.hpp"
+#include "storage/storage_manager.hpp"
+#include "storage/table.hpp"
+#include "utils/logging.hpp"
+
+namespace opossum {
+
+void IndexOperation::execute() {
+  if (_create) {
+    _create_index();
+  } else {
+    _delete_index();
+  }
+}
+
+void IndexOperation::print_on(std::ostream& output) const {
+  output << "IndexOperation{" << (_create ? "Create" : "Delete") << " on " << _column << "}";
+}
+
+void IndexOperation::_create_index() {
+  auto table = StorageManager::get().get_table(_column.table_name);
+  switch (_type) {
+    case ColumnIndexType::GroupKey:
+      table->create_index<GroupKeyIndex>(_column.column_ids);
+      break;
+    case ColumnIndexType::CompositeGroupKey:
+      table->create_index<CompositeGroupKeyIndex>(_column.column_ids);
+      break;
+    case ColumnIndexType::AdaptiveRadixTree:
+      table->create_index<AdaptiveRadixTreeIndex>(_column.column_ids);
+      break;
+    default:
+      Fail("Can not create invalid/unknown index type");
+  }
+
+  _invalidate_cache();
```

Comment:
Do we clear the cache for each creation/deletion of an index? Why not do it after all operations completed? Just a quick thought: Could we theoretically store for each choice the cache entries that influenced this choice and only remove these?

Reply:
Unfortunately, we do that at the moment. And yes, that would be theoretically possible - if the cache had an interface to selectively evict entries. That would be our preferred method.
For now, we could circumvent clearing the cache for each index operation by implementing a kind of post-process-hook system. Should we do this? Or do you prefer the more specific former approach?

Reply:
Ideally, we would only evict plans that are useless after tuning but if that is too much effort we should at least only clear the cache once after the whole tuning process.

Reply:
As discussed: We will create an e2e test that tests the whole tuning system. In that test scope, it will clear the cache manually after the tuning operations are completed and we'll add a remark that (a) the cache may contain suboptimal queries after tuning and (b) whoever uses the tuner for tuning indices is responsible for clearing the cache afterwards.